opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -553,6 +553,19 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
             with self._lock:
                 self.attributes[key] = value
 
+    def get_attribute(self, key: str) -> types.AttributeValue:
+        if key is None:
+            logger.warning("invalid key (empty or null)")
+            return None
+
+        with self._lock:
+            if key not in self.attributes:
+                logger.warning("no attributes for key:%s", str(key))
+                return None
+
+        with self._lock:
+            return self.attributes[key]
```

Comment:
Would be better to do this under a single lock acquisition. As written here, you release the lock after the check on L562, then another thread could acquire the lock, and we have to block again until it's released on L566.